src/Language/PureScript/Pretty/JS.hs

```diff
@@ -58,9 +57,9 @@ literals = mkPattern' match'
     , return $ emit "}"
     ]
     where
-    objectPropertyToString :: (Emit gen) => Text -> gen
-    objectPropertyToString s | identNeedsEscaping s = string s
-                             | otherwise = emit s
+    objectPropertyToString :: (Emit gen) => PSString -> gen
+    objectPropertyToString s | identNeedsEscaping $ T.pack $ codePoints s = string s
```

Comment:
Ok, cool. Is there any part of that where we need the current behaviour of `codePoints`, i.e. representing lone surrogates in that way?